### PR TITLE
Points merging

### PR DESCRIPTION
**Pull request by RaphaelBonatti**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/artificial_intelligence/ArtificialMovingEntity.java [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
-import ch.epfl.sdp.geometry.Positionable;
+import ch.epfl.sdp.geometry.LocalArea;
 import ch.epfl.sdp.geometry.UnboundedArea;
 
 /**
  * Represents an entity of the game that can move automatically by setting a movement. The movement
  * can be limited by setting an area where the entity can reside.
  */
-public abstract class MovingArtificialEntity extends AoeRadiusMovingEntity implements Movable, Positionable, Updatable {
+public abstract class ArtificialMovingEntity extends AoeRadiusEntity implements Updatable {
```
Comment (AmroAbdrabo) [inline app/src/main/java/ch/epfl/sdp/artificial_intelligence/ArtificialMovingEntity.java:17]: Good!

File: app/src/main/java/ch/epfl/sdp/database/firebase/entity/EnemyForFirebase.java [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-import ch.epfl.sdp.geometry.GeoPoint;
+import ch.epfl.sdp.database.firebase.GeoPointForFirebase;
 
 /**
  * The in-game enemy entity to be stored in the cloud firebase
  */
 public class EnemyForFirebase {
     private int id;
-    private GeoPoint location;
+    private GeoPointForFirebase location;
```
Comment (AmroAbdrabo) [inline app/src/main/java/ch/epfl/sdp/database/firebase/entity/EnemyForFirebase.java:10]: I think a good idea for the future is to have on class that adapts everything we need for Firebase (that's what other teams did that had 90% coverage). I'm not going to make you do it on this pull request but it's a good idea for a later pull request

File: app/src/main/java/ch/epfl/sdp/database/firebase/GeoPointForFirebase.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package ch.epfl.sdp.database.firebase;
```
[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/database/firebase/GeoPointForFirebase.java:1]: The suggestion from Amro will also solve all those code climate issue

[the remark this answers is not in the document]
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/database/firebase/GeoPointForFirebase.java:1]: (I think)

File: app/src/main/java/ch/epfl/sdp/entity/Enemy.java [hunk 4]
```diff
@@ -253,9 +249,9 @@ private void chase() {
                 behaviour = Behaviour.ATTACK;
             }
         } else {
-            super.setArea(patrolBounds);
+            super.setLocalArea(patrolBounds);
             setForceMove(true);
-            super.getMovement().setVelocity(super.getMovement().getVelocity() / 2);
+            //super.getMovement().setVelocity(super.getMovement().getVelocity() / 2);
```
Comment (Zeavee) [inline app/src/main/java/ch/epfl/sdp/entity/Enemy.java:254]: Do not hesitate to delete lines :)

File: app/src/main/java/ch/epfl/sdp/geometry/GeoPoint.java [hunk 5]
```diff
@@ -80,4 +69,51 @@ private double haversin(double x) {
...
+     * Gets the point's y coordinate.
+     *
+     * @return A value representing the y coordinate.
+     */
+    public double getY() {
+        UTMRef utm = geoPointToUTMRef(this);
+        return utm.getNorthing();
+    }
+
+    public GeoPoint asOriginTo(Vector vector) {
```
Comment (Zeavee) [inline app/src/main/java/ch/epfl/sdp/geometry/GeoPoint.java:93]: Missing some javadoc for this method

File: app/src/main/java/ch/epfl/sdp/database/firebase/entity/EnemyForFirebase.java (hunk 2 above)
Comment (SkyRiS3s) [reply app/src/main/java/ch/epfl/sdp/database/firebase/entity/EnemyForFirebase.java:10]: Yeah I agree with Amro as well

**Pull request merged**